### PR TITLE
Delete segments after upload if upload is enabled

### PR DESCRIPTION
**Pull request by biglittlebigben**
Is this safe to do?

### CODE REVIEW

**Comment by frostbyte73** [comment]
`*Uploader` is the uploader base, it should clean up in uploader.go L61.
We'll also need to add a parameter to upload, because we don't want to delete the playlist after uploading but everything else should be ok

File: pkg/pipeline/sink/uploader/uploader.go [hunk 1]
```diff
@@ -89,6 +94,17 @@ func (u *noOpUploader) upload(localFilepath, _ string, _ types.OutputType) (stri
...
 }
 
+func (u *noOpUploader) cleanupFile(localFilepath string) error {
+	return nil
+}
+
+type baseUploader struct {
+}
+
+func (u *baseUploader) cleanupFile(localFilepath string) error {
```
Comment (frostbyte73) [inline pkg/pipeline/sink/uploader/uploader.go:104]: `func (u *Uploader) cleanupFile(localFilepath string) error {`

Comment (biglittlebigben) [reply pkg/pipeline/sink/uploader/uploader.go:104]: Wouldn't the cause the file to be deleted as well in cases where there is no cloud upload (which I believe means that the noopUploader is selected)?

Comment (frostbyte73) [reply pkg/pipeline/sink/uploader/uploader.go:104]: oh, true. In that case, better to kill `*Uploader` and change the `uploader` interface to be public. 
Then the backup logic can move into `baseUploader`

**Review by frostbyte73 — APPROVED** [review]

**Pull request merged**